virtool/subtractions/db.py

```diff
@@ -79,14 +85,33 @@ async def add_files_field(self):
         await virtool.tasks.pg.update(
             self.pg,
             self.id,
-            step="add_files_field"
+            step="store_subtraction_files"
         )
 
         async for subtraction in self.db.subtraction.find(ADD_SUBTRACTION_FILES_QUERY):
-            files = await virtool.subtractions.utils.prepare_files_field(self.pg, subtraction["_id"])
+            path = virtool.subtractions.utils.join_subtraction_path(settings, subtraction["_id"])
+            subtraction_files = list()
+
+            for filename in sorted(os.listdir(path)):
+                if filename in FILES:
+                    async with AsyncSession(self.app["pg"]) as session:
+                        exists = (await session.execute(
+                            select(SubtractionFile).filter_by(subtraction=subtraction["_id"], name=filename)
+                        )).scalar()
+
+                    if not exists:
+                        subtraction_files.append(filename)
+
+            await virtool.subtractions.files.create_subtraction_files(
+                self.app["pg"],
+                subtraction["_id"],
+                subtraction_files,
+                path
+            )
+
             await self.db.subtraction.update_one({"_id": subtraction["_id"]}, {
-                "$set": {
-                    "files": files
+                "$unset": {
+                    "files": ""
                 }
             })
```

Comment:
On second thought. Let's just leave the `files` field in case there is a problem with the SQL switch.